classification/feature_generation/feature_utilities.py

```diff
@@ -0,0 +1,373 @@
+# Copyright 2017 Google Inc. and Skytruth Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import pytz
+import logging
+import numpy as np
+import time
+
+
+EPOCH_DT = datetime.datetime(1970, 1, 1, tzinfo=pytz.utc)
+
+
+def np_pad_repeat_slice(slice, window_size):
+    """ Pads slice to the specified window size.
+
+  Series that are shorter than window_size are repeated into unfilled space.
+
+  Args:
+    slice: a numpy array.
+    window_size: the size the array must be padded to.
+
+  Returns:
+    a numpy array of length window_size in the first dimension.
+  """
+
+    slice_length = len(slice)
+    assert (slice_length <= window_size)
+    reps = int(np.ceil(window_size / float(slice_length)))
+    return np.concatenate([slice] * reps, axis=0)[:window_size]
+
+def np_zero_pad_slice(slice, window_size, random_state):
+    """ Pads slice to the specified window size.
+
+  Series that are shorter than window_size are repeated into unfilled space.
+
+  Args:
+    slice: a numpy array.
+    window_size: the size the array must be padded to.
+
+  Returns:
+    a numpy array of length window_size in the first dimension.
+  """
+
+    slice_length = len(slice)
+    delta = window_size - slice_length
+    assert delta >= 0
+    _, n_feat = slice.shape
+    padded = np.zeros([window_size, n_feat], dtype=slice.dtype)
+    offset = random_state.randint(0, delta + 1)
+    return np.concatenate([slice] * reps, axis=0)[offset:offset+window_size]
+
+def np_pad_repeat_slice_2(slice, window_size, random_state):
+    """ Pads slice to the specified window size.
+
+  Series that are shorter than window_size are repeated into unfilled space.
```

Comment:
Would you mind adding an extra sentence of description for how this differs from the above similarly named fn?